### PR TITLE
Improve service URL handling

### PR DESCRIPTION
**Pull request by zaneb**
Restore IPv6 support, pick node0 more reliably, and have a single source of truth for the service URL.

### CODE REVIEW

File: data/ignition/files/usr/local/bin/start-cluster-installation.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@ source common.sh
 
 wait_for_assisted_service
 
-BASE_URL="http://{{.NodeZeroIP}}:8090/api/assisted-install/v2"
+BASE_URL="{{.ServiceBaseURL}}/api/assisted-install/v2"
```
Comment (pawanpinjarkar) [inline data/ignition/files/usr/local/bin/start-cluster-installation.sh]: This results in an extra `/` as `BASE_URL="http://192.168.111.80:8090//api/assisted-install/v2`

File: pkg/imagebuilder/content.go [hunk 2]
```diff
@@ -41,8 +41,11 @@ func New() *ConfigBuilder {
 	// TODO: needs appropriate value if AUTH_TYPE != none
 	pullSecretToken := getEnv("PULL_SECRET_TOKEN", "")
 
-	serviceBaseURL := fmt.Sprintf("http://%s/",
-		net.JoinHostPort(nodeZeroIP, "8090"))
+	serviceBaseURL := url.URL{
+		Scheme: "http",
+		Host:   net.JoinHostPort(nodeZeroIP, "8090"),
+		Path:   "/",
```
Comment (pawanpinjarkar) [inline pkg/imagebuilder/content.go:47]: Looks like this adds extra `/` and results in `BASE_URL="http://192.168.111.80:8090//api/assisted-install/v2"` in `/usr/local/bin/start-cluster-installation.sh  which then breaks auto-start cluster installation.
https://github.com/openshift-agent-team/fleeting/blob/main/data/ignition/files/usr/local/bin/start-cluster-installation.sh#L8

Comment (pawanpinjarkar) [inline pkg/imagebuilder/content.go:47]: Looks like this adds extra `/` and results in `BASE_URL="http://192.168.111.80:8090//api/assisted-install/v2"` in `/usr/local/bin/start-cluster-installation.sh  which then breaks auto-start cluster installation.
https://github.com/openshift-agent-team/fleeting/blob/main/data/ignition/files/usr/local/bin/start-cluster-installation.sh#L8

**Review by pawanpinjarkar — CHANGES_REQUESTED** [review]

File: data/ignition/files/usr/local/bin/start-cluster-installation.sh (hunk 1 above)
Comment (zaneb) [reply data/ignition/files/usr/local/bin/start-cluster-installation.sh]: Fixed, thanks.

**Pull request merged**